jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -0,0 +1,124 @@
+
+    def _sync_available_from_state(self, *args):
+        self.subset_items = [{'label': self.no_selection_text}] + [
+                             self.subset_select._subset_to_dict(subset) for subset in
+                             self.data_collection.subset_groups]
+
+    @observe('subset_selected')
+    def _sync_selected_from_ui(self, change):
+        if change['new'] != self.no_selection_text:
+            self._get_region_definition(change['new'])
+        self.show_region_info = change['new'] != self.no_selection_text
+        m = [s for s in self.app.data_collection.subset_groups if s.label == change['new']]
+        if m != self.session.edit_subset_mode.edit_subset:
+            self.session.edit_subset_mode.edit_subset = m
+
+    '''
+    # This will be needed once we use a dropdown instead of the actual
+    # g-subset-mode component
+    @observe("mode_selected")
+    def _mode_selected_changed(self, event={}):
+        if self.session.edit_subset_mode != self.mode_selected:
+            self.session.edit_subset_mode = self.mode_selected
+    '''
+
+    def _get_region_definition(self, *args):
+        self.subset_definition = {}
+        subset_group = [s for s in self.app.data_collection.subset_groups if
+                        s.label == self.subset_selected][0]
+        subset_state = subset_group.subset_state
+        subset_class = subset_state.__class__
+
+        if subset_class in (OrState, AndState, XorState, InvertState):
+            self.subset_classname = "Compound Subset"
+        else:
+            if isinstance(subset_state, RoiSubsetState):
+                self.subset_classname = subset_state.roi.__class__.__name__
+                if self.subset_classname == "CircularROI":
+                    x, y = subset_state.roi.get_center()
+                    self.subset_definition = {"X Center": x,
+                                              "Y Center": y,
+                                              "Radius": subset_state.roi.radius}
+                elif self.subset_classname == "RectangularROI":
+                    temp_def = {}
+                    for att in ("Xmin", "Xmax", "Ymin", "Ymax"):
+                        temp_def[att] = getattr(subset_state.roi, att.lower())
+                    self.subset_definition = temp_def
+                elif self.subset_classname == "EllipticalROI":
+                    self.subset_definition = {"X Center": subset_state.roi.xc,
+                                              "Y Center": subset_state.roi.yc,
+                                              "X Radius": subset_state.roi.radius_x,
+                                              "Y Radius": subset_state.roi.radius_y}
+            elif isinstance(subset_state, RangeSubsetState):
+                self.subset_classname = "Range"
+                self.subset_definition = {"Upper bound": subset_state.hi,
+                                          "Lower bound": subset_state.lo}
```

Comment:
should there be a final else here that catches other scenarios or sets `subset_classname='unknown'`?

Reply:
I debated this as well, I had one originally while debugging but deleted it - this is still getting triggered once on subset creation when they're still bare `SubsetState` objects (and then immediately updated to the correct information), I don't want to flicker something when that happens. But I do see that Imviz uses `MaskSubsetState` (for the static regions I assume), and I wonder if checking rows in Mosviz ends up with an `ElementSubsetState`...I'll go ahead and add an `else` case.